### ISSUE TITLE
Build the SSH2 extension against a more recent version of libssh2

### ISSUE THREAD

**Issue by mignaulo** [oerdnj/deb.sury.org#1368]
**Frequently asked questions**

- [ X ] I have read [Frequenty Asked Questions](https://github.com/oerdnj/deb.sury.org/wiki/Frequently-Asked-Questions)

**Is your feature request related to a problem? Please describe.**
We are trying to use the SSH2 extension to send commands to various other servers over SSH. This has worked well. We now have servers running on Ubuntu Bionic. The more recent version of OpenSSH has stricter requirements for key exchange. PHP cannot communicate with this server over SSH, because there is no key exchange in common between libssh2 1.5 (the version used here is rather old, it only supports Diffie-Hellman SHA1 while OpenSSH explicitely disables it) and Ubuntu Bionic. 

**Describe the solution you'd like**
If the SSH2 extension is built at least against libssh2 version 1.7, there will be key exchange options in common between php_ssh2 and more modern servers.

**Describe alternatives you've considered**
There are some workarounds here: 
* It's possible to allow again DH SHA1 on the new Bionic server, but I'd prefer not to enable features that have been disabled for security.
* phpseclib (pure PHP implementation) seems to support the newer key exchange options like DH SHA256. I believe this would work, probably with reduced performance.

**Distribution (please complete the following information):**
 - OS: Ubuntu 16.04(xenial)/18.04 (bionic)
 - Architecture: amd64
 - Repository: packages.sury.org

**Package(s) (please complete the following information):**
php-ssh2: 1.2+0.13-1+ubuntu16.04.1+deb.sury.org+1
```
ssh2

SSH2 support => enabled
extension version => 0.13
libssh2 version => 1.5.0
banner => SSH-2.0-libssh2_1.5.0
```

Thank you for considering this!

**Comment by stof** [comment]
Well, libssh2 1.5.0 is the version shipped in Ubuntu 16.04. Ubuntu 18.04 ships with libss2 1.8.0. Is the compilation actually done against 1.5.0 also for your 18.04 setup (your issue mentions both versions) ? For me, I'm getting 1.8.0 when I try that. So the answer seems simple: upgrade to bionic :smile:

**Comment by mignaulo** [comment]
I had done some tests and I was certain bionic also used libssh2 1.5.0 for some reason.
Glad that it's not the case. Closing this.
